src/Swarm/TUI/View.hs

```diff
@@ -326,9 +326,26 @@ renderErrorDialog err = renderDialog (dialog (Just "Error") Nothing (maxModalWin
 -- | Draw the error dialog window, if it should be displayed right now.
 drawDialog :: AppState -> Widget Name
 drawDialog s = case s ^. uiState . uiModal of
-  Just (Modal mt d) -> renderDialog d (withVScrollBars OnRight $ drawModal s mt)
+  Just (Modal mt d) -> renderDialog d (maybeScroll ModalViewport $ drawModal s mt)
   Nothing -> maybe emptyWidget renderErrorDialog (s ^. uiState . uiError)
 
+-- | Make a widget scrolling if it is bigger than the available
+--   vertical space.  Thanks to jtdaugherty for this code.
+maybeScroll :: (Ord n, Show n) => n -> Widget n -> Widget n
+maybeScroll vpName contents =
```

Comment:
Sumptuous! 🧐 (I like it 😉)

Reply:
Haha, thanks! Though I can't claim much credit. :smile: